### PR TITLE
Fixes #3811: Indicate vocab name when editing term

### PR DESCRIPTION
**Pull request by laryn**
Fixes #3811

### CODE REVIEW

File: core/modules/taxonomy/taxonomy.admin.inc [hunk 1]
```diff
@@ -738,27 +738,24 @@ function taxonomy_form_term($form, &$form_state, TaxonomyTerm $term = NULL, Taxo
...
+    $vocabulary = taxonomy_vocabulary_load($term->vocabulary);
+  }
   if (!isset($form_state['term'])) {
     // Create a new TaxonomyTerm entity for the add form.
     if (!isset($term)) {
-      backdrop_set_title(t('Add term'));
       $term = entity_create('taxonomy_term', array(
         'vocabulary' => $vocabulary->machine_name,
       ));
+      backdrop_set_title(t('Add @vocab_name term', array('@vocab_name' => $vocabulary->name)), PASS_THROUGH);
```
Comment (klonos) [inline core/modules/taxonomy/taxonomy.admin.inc]:
```suggestion
      backdrop_set_title(t('Add term to the %vocab_name vocabulary', array('%vocab_name' => $vocabulary->name)), PASS_THROUGH);
```

File: core/modules/taxonomy/taxonomy.admin.inc [hunk 2]
```diff
@@ -738,27 +738,24 @@ function taxonomy_form_term($form, &$form_state, TaxonomyTerm $term = NULL, Taxo
...
     }
     else {
-      backdrop_set_title(t('Edit term %term_name', array('%term_name' => $term->name)), PASS_THROUGH);
-    }
-    if (!isset($vocabulary) && isset($term->vocabulary)) {
-      $vocabulary = taxonomy_vocabulary_load($term->vocabulary);
+      backdrop_set_title(t('Edit @vocab_name term %term_name', array('@vocab_name' => $vocabulary->name, '%term_name' => $term->name)), PASS_THROUGH);
```
Comment (klonos) [inline core/modules/taxonomy/taxonomy.admin.inc]:
```suggestion
      backdrop_set_title(t('Edit term %term_name in the @vocab_name vocabulary', array('@vocab_name' => $vocabulary->name, '%term_name' => $term->name)), PASS_THROUGH);
```

File: core/modules/taxonomy/taxonomy.admin.inc [hunk 3]
```diff
@@ -738,27 +738,24 @@ function taxonomy_form_term($form, &$form_state, TaxonomyTerm $term = NULL, Taxo
   // During initial form build, add the term entity to the form state for use
   // during form building and processing. During a rebuild, use what is in the
   // form state.
+  if (!isset($vocabulary) && isset($term->vocabulary)) {
+    $vocabulary = taxonomy_vocabulary_load($term->vocabulary);
+  }
   if (!isset($form_state['term'])) {
```
Comment (ghost) [inline core/modules/taxonomy/taxonomy.admin.inc]: I think the `if` should go above the "During initial form build..." comment, with its own comment above it saying something like: "Load the vocabulary from the term, if not already provided." And then put a blank line between this `if` function and the "During initial form build..." comment.

```suggestion
  // Load the vocabulary from the term, if not already provided.
  if (!isset($vocabulary) && isset($term->vocabulary)) {
    $vocabulary = taxonomy_vocabulary_load($term->vocabulary);
  }
  
  // During initial form build, add the term entity to the form state for use
  // during form building and processing. During a rebuild, use what is in the
  // form state.
  if (!isset($form_state['term'])) {
```

**Review by ghost — CHANGES_REQUESTED** [review]

File: core/modules/taxonomy/taxonomy.admin.inc (hunk 2 above)
Comment (laryn) [reply core/modules/taxonomy/taxonomy.admin.inc]: I adjusted using alternate text per the issue discussion.

File: core/modules/taxonomy/taxonomy.admin.inc (hunk 1 above)
Comment (laryn) [reply core/modules/taxonomy/taxonomy.admin.inc]: I adjusted using alternate text per the issue discussion.

**Review by klonos — APPROVED** [review]
LGTM 👍

**Pull request merged**